subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/TaskNodeCodec.kt

```diff
@@ -287,7 +288,7 @@ suspend fun WriteContext.writeRegisteredPropertiesOf(
     val outputProperties = collectRegisteredOutputsOf(task)
     writeCollection(outputProperties) { property ->
         property.run {
-            val finalValue = DeferredUtil.unpack(propertyValue)
+            val finalValue = DeferredUtil.unpack(AbsentProviderHandling.ALLOW_ABSENT, propertyValue)
```

Comment:
WDYT about keeping `Object unpack()` to works as it does today (=strict), and add an `unpackIfPresent(Consumer<Object>)` for the lenient case?

Reply:
I did that.

Reply:
What you did is somewhat different to what I suggested (which is to handle the unpacked value via a `Consumer`). Your approach could also work, but if we go with it, I think the method should be called `unpackOrNull()` instead.

Reply:
Right. I renamed the method to `unpackOrNull()`. The consumer variant doesn't work really well, since I mostly need to do something in the `null` case as well.